### PR TITLE
fix: Replace default file_cache usage with simple memory cache

### LINKED ISSUE #4185: ImportError shows in logs when implementing datastore export sample
## In which file did you encounter the issue?

datastore/schedule-export/main.py

### Did you change the file? If so, how?

No

## Describe the issue

When implementing a cloud function using schedule-export/main.py, these messages show up in the logs on a run (the function still executes successfully):

```
2020-06-26T19:34:03.459Z datastore_export file_cache is unavailable when using oauth2client >= 4.0.0 or google-auth E  datastore_export 
2020-06-26T19:34:03.459Z datastore_export Traceback (most recent call last): E  datastore_export 
2020-06-26T19:34:03.459Z datastore_export   File "/env/local/lib/python3.7/site-packages/googleapiclient/discovery_cache/file_cache.py", line 33, in <module> E  datastore_export 
2020-06-26T19:34:03.459Z datastore_export     from oauth2client.contrib.locked_file import LockedFile E  datastore_export 
2020-06-26T19:34:03.459Z datastore_export ModuleNotFoundError: No module named 'oauth2client' E  datastore_export 
2020-06-26T19:34:03.459Z datastore_export  E  datastore_export 
2020-06-26T19:34:03.459Z datastore_export During handling of the above exception, another exception occurred: E  datastore_export 
2020-06-26T19:34:03.459Z datastore_export  E  datastore_export 
2020-06-26T19:34:03.459Z datastore_export Traceback (most recent call last): E  datastore_export 
2020-06-26T19:34:03.459Z datastore_export   File "/env/local/lib/python3.7/site-packages/googleapiclient/discovery_cache/file_cache.py", line 37, in <module> E  datastore_export 
2020-06-26T19:34:03.459Z datastore_export     from oauth2client.locked_file import LockedFile E  datastore_export 
2020-06-26T19:34:03.459Z datastore_export ModuleNotFoundError: No module named 'oauth2client' E  datastore_export 
2020-06-26T19:34:03.459Z datastore_export  E  datastore_export 
2020-06-26T19:34:03.459Z datastore_export During handling of the above exception, another exception occurred: E  datastore_export 
2020-06-26T19:34:03.459Z datastore_export  E  datastore_export 
2020-06-26T19:34:03.459Z datastore_export Traceback (most recent call last): E  datastore_export 
2020-06-26T19:34:03.459Z datastore_export   File "/env/local/lib/python3.7/site-packages/googleapiclient/discovery_cache/__init__.py", line 44, in autodetect E  datastore_export 
2020-06-26T19:34:03.459Z datastore_export     from . import file_cache E  datastore_export 
2020-06-26T19:34:03.459Z datastore_export   File "/env/local/lib/python3.7/site-packages/googleapiclient/discovery_cache/file_cache.py", line 41, in <module> E  datastore_export 
2020-06-26T19:34:03.459Z datastore_export     "file_cache is unavailable when using oauth2client >= 4.0.0 or google-auth" E  datastore_export 
2020-06-26T19:34:03.459Z datastore_export ImportError: file_cache is unavailable when using oauth2client >= 4.0.0 or google-auth E  datastore_export 
```

The error referred to is mentioned in this issue, which suggests a workaround: https://github.com/googleapis/google-api-python-client/issues/325#issuecomment-274349841

### PR DESCRIPTION
**Pull request by aawilson**
The original main.py creates noisy log messages when the function is run file_cache is unavailable when using oauth2client >= 4.0.0:

```
2020-06-26T19:34:03.459Z datastore_export file_cache is unavailable when using oauth2client >= 4.0.0 or google-auth E  datastore_export 
2020-06-26T19:34:03.459Z datastore_export Traceback (most recent call last): E  datastore_export 
2020-06-26T19:34:03.459Z datastore_export   File "/env/local/lib/python3.7/site-packages/googleapiclient/discovery_cache/file_cache.py", line 33, in <module> E  datastore_export 
2020-06-26T19:34:03.459Z datastore_export     from oauth2client.contrib.locked_file import LockedFile E  datastore_export 
2020-06-26T19:34:03.459Z datastore_export ModuleNotFoundError: No module named 'oauth2client' E  datastore_export 
2020-06-26T19:34:03.459Z datastore_export  E  datastore_export 
2020-06-26T19:34:03.459Z datastore_export During handling of the above exception, another exception occurred: E  datastore_export 
2020-06-26T19:34:03.459Z datastore_export  E  datastore_export 
2020-06-26T19:34:03.459Z datastore_export Traceback (most recent call last): E  datastore_export 
2020-06-26T19:34:03.459Z datastore_export   File "/env/local/lib/python3.7/site-packages/googleapiclient/discovery_cache/file_cache.py", line 37, in <module> E  datastore_export 
2020-06-26T19:34:03.459Z datastore_export     from oauth2client.locked_file import LockedFile E  datastore_export 
2020-06-26T19:34:03.459Z datastore_export ModuleNotFoundError: No module named 'oauth2client' E  datastore_export 
2020-06-26T19:34:03.459Z datastore_export  E  datastore_export 
2020-06-26T19:34:03.459Z datastore_export During handling of the above exception, another exception occurred: E  datastore_export 
2020-06-26T19:34:03.459Z datastore_export  E  datastore_export 
2020-06-26T19:34:03.459Z datastore_export Traceback (most recent call last): E  datastore_export 
2020-06-26T19:34:03.459Z datastore_export   File "/env/local/lib/python3.7/site-packages/googleapiclient/discovery_cache/__init__.py", line 44, in autodetect E  datastore_export 
2020-06-26T19:34:03.459Z datastore_export     from . import file_cache E  datastore_export 
2020-06-26T19:34:03.459Z datastore_export   File "/env/local/lib/python3.7/site-packages/googleapiclient/discovery_cache/file_cache.py", line 41, in <module> E  datastore_export 
2020-06-26T19:34:03.459Z datastore_export     "file_cache is unavailable when using oauth2client >= 4.0.0 or google-auth" E  datastore_export 
2020-06-26T19:34:03.459Z datastore_export ImportError: file_cache is unavailable when using oauth2client >= 4.0.0 or google-auth E  datastore_export 
```

While they don't interfere with the function's operation, they are disconcerting and not necessary. This implements the workaround suggested in https://github.com/googleapis/google-api-python-client/issues/325 (which creates an in-memory cache object to use in replacement).

## Description

Fixes #4185 

## Checklist
- [x] I have followed [Sample Guidelines from AUTHORING_GUIDE.MD](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/AUTHORING_GUIDE.md)
- [x] README is updated to include [all relevant information](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/AUTHORING_GUIDE.md#readme-file)
- [ ] **Tests** pass:   `nox -s py-3.6` (see [Test Enviroment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] **Lint** pass:   `nox -s lint` (see [Test Enviroment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/AUTHORING_GUIDE.md#test-environment-setup))
- [ ] These samples need a new **API enabled** in testing projects to pass (let us know which ones)
- [ ] These samples need a new/updated **env vars** in testing projects set to pass (let us know which ones)
- [x] Please **merge** this PR for me once it is approved.

### CODE REVIEW

**Comment by leahecole** [comment]
PTAL @BenWhitehead or @crwilcox to see how you feel about it from a Datastore point of view

File: datastore/schedule-export/main.py [hunk 1]
```diff
@@ -3,8 +3,20 @@
...
+    _CACHE = {}
+
+    def get(self, url):
+        return MemoryCache._CACHE.get(url)
+
+    def set(self, url, content):
+        MemoryCache._CACHE[url] = content
+
+
+datastore = build('datastore', 'v1', cache=MemoryCache())
```
Comment (crwilcox) [inline datastore/schedule-export/main.py:19]: Can we add a comment here to explain adding this?

```
Provide a `MemoryCache` instance as file_cache is unavailable when using oauth2client >= 4.0.0 or google-auth
```

Comment (aawilson) [reply datastore/schedule-export/main.py:20]: Have done, added a bit more to explain the "why" as well.

**Comment by leahecole** [comment]
@aawilson it looks like this is failing lint - there needs to be one more blank line in main.py line 20. You can also test this locally with `nox -s lint` (see [Test Enviroment Setup](https://github.com/GoogleCloudPlatform/python-docs-samples/blob/master/AUTHORING_GUIDE.md#test-environment-setup))

**Comment by aawilson** [comment]
Yup got it, my local linter would have caught it but I've been editing this entirely in GitHub's interface, will add

**Comment by leahecole** [comment]
@aawilson I've been there! Thanks for the edits 😄

**Pull request merged**